libraries/lib-files/FileNames.cpp

```diff
@@ -245,23 +288,45 @@ FilePath FileNames::DataDir()
 #endif
       wxFileName portablePrefsPath(exePath.GetPath(), wxT("Portable Settings"));
 
+#if defined(__WXGTK__)
+      if (gOldUnixDataDir.empty())
+         gOldUnixDataDir = wxFileName::GetHomeDir() + wxT("/.audacity-data");
+#endif
+
       if (::wxDirExists(portablePrefsPath.GetFullPath()))
       {
          // Use "Portable Settings" folder
-         gDataDir = portablePrefsPath.GetFullPath();
+         dir = portablePrefsPath.GetFullPath();
+#if defined(__WXGTK__)
+      } else if (::wxDirExists(gOldUnixDataDir))
+      {
+         // Use old user data dir folder
+         dir = gOldUnixDataDir;
+      } else
+      {
+         wxString newDir(xdgTargetDir(target));
+#ifdef AUDACITY_NAME
+         newDir = newDir + wxT("/" AUDACITY_NAME);
+#else
+         newDir = newDir + wxT("/audacity");
+#endif
+#else
       } else
       {
          // Use OS-provided user data dir folder
-         wxString dataDir( LowerCaseAppNameInPath( wxStandardPaths::Get().GetUserDataDir() ));
-#if defined( __WXGTK__ )
-         dataDir = dataDir + wxT("-data");
+         wxString newDir( FileNames::LowerCaseAppNameInPath( wxStandardPaths::Get().GetUserDataDir() ));
 #endif
-         gDataDir = FileNames::MkDir(dataDir);
+         dir = FileNames::MkDir(newDir);
```

Comment:
Looks like you want overwrite the value, so you need `auto& dir = gTargetDir(target);`